### PR TITLE
fixing GPU memory leak in order_1_approx

### PR DESCRIPTION
**Pull request by chanind**
I find then when I run `scripts.sweep`, I get OOM errors. It looks like there's an explicit OOM error handling path in the code, suggesting this is a common occurrence for running this script.

It looks like this is related to the `order_1_approx()` function, and I find that by adding `.detach().clone()` to the `weight` and `bias` it seems to alleviate the issue and I no longer get the OOM errors. I suspect that the jacobian calculation may be leaving behind a computation graph on the GPU that doesn't get freed, and calling `.detach().clone()` fully separates the returned tensors from the computation graph allowing garbage collection to clean up GPU memory. This is just my guess though, and I haven't done a memory profile to prove that's what going on.

### CODE REVIEW

**Comment by chanind** [comment]
I may have been too optimistic about this - I find that I still get occasional OOM errors with this PR, although they're much less frequent. Given that, I'm not sure if it makes sense to merge this or not as it doesn't seem to 100% solve the issue.

**Comment by chanind** [comment]
I think this fix should be correct now. I somehow overlooked the final tensor return value `Jh` in `metadata`. I updated the PR so this value as well has `.detach().clone()` called on it. I've been running the sweep script on a 24gb GPU for the past hour with no memory issues, so *hopefully* this means the leak is fixed.

**Comment by chanind** [comment]
closing for now as I'm not convinced this is fixing the core issue
